server/system/service/smtp_connection_checker.go

```diff
@@ -0,0 +1,159 @@
+package service
+
+import (
+	"context"
+	"fmt"
+	"github.com/cortezaproject/corteza/server/pkg/actionlog"
+	intAuth "github.com/cortezaproject/corteza/server/pkg/auth"
+	"github.com/cortezaproject/corteza/server/pkg/mail"
+	"github.com/cortezaproject/corteza/server/pkg/options"
+	"github.com/cortezaproject/corteza/server/system/types"
+	gomail "gopkg.in/mail.v2"
+	htpl "html/template"
+	"io/ioutil"
+	"strings"
+)
+
+type (
+	smtpConnectionChecker struct {
+		actionlog actionlog.Recorder
+		settings  *types.AppSettings
+		ts        TemplateService
+		opt       options.AuthOpt
+	}
+)
+
+func SmtpConnectionChecker(s *types.AppSettings, ts TemplateService, opt options.AuthOpt) *smtpConnectionChecker {
+	return &smtpConnectionChecker{
+		actionlog: DefaultActionlog,
+		settings:  s,
+		ts:        ts,
+		opt:       opt,
+	}
+}
+
+// SmtpServerChecker Expected to error out
+func (svc smtpConnectionChecker) SmtpServerChecker(ctx context.Context, smtpConfigs *types.SmtpServer) (err error) {
+	//check for validity of the host
+	if !mail.IsValidHost(smtpConfigs.Host) {
+		err = fmt.Errorf("%s name is Invalid", smtpConfigs.Host)
+	}
+
+	if err != nil {
+		return err
+	}
+
+	//check the validity of the port
+	if !mail.IsValidPort(smtpConfigs.Port) {
+		err = fmt.Errorf("%s Port number is Invalid", smtpConfigs.Port)
+	}
+
+	if err != nil {
+		return err
+	}
+
+	err = mail.SMTPServerCheck(smtpConfigs.Host, smtpConfigs.Port, smtpConfigs.Username, smtpConfigs.Password)
+	if err != nil {
+		return err
+	}
+
+	//send the email
+	err = svc.smtpSendEmail(ctx, smtpConfigs.Recipients)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (svc smtpConnectionChecker) smtpSendEmail(ctx context.Context, recipients []string) (err error) {
+	var (
+		tmp []byte
+		ntf = mail.New()
+		hdl string
+		// context with service user
+		// we need this for retrieving & rendering email templates
+		suCtx = intAuth.SetIdentityToContext(ctx, intAuth.ServiceUser())
```

Comment:
Kudos on this!! First I thought you missed the scenario where user could be w/o permissions to check templates. Kudos!